packages/crayons-icon/utils/svgo.runner.js

```diff
@@ -0,0 +1,117 @@
+/* eslint-disable no-undef */
+const fs_extra = require('fs-extra');
+const path = require('path');
+const svgo = require('svgo');
+const log = require('./log');
+
+module.exports = build;
+async function build(rootDir, srcDir, distDir, loadConfigPlugins) {
+  try {
+    let alert = {};
+    const srcSvgDir = path.join(rootDir, srcDir);
+    const distSvgDir = path.join(rootDir, distDir);
+
+    alert = {
+      type: 'info',
+      name: 'Optimizing SVGs In-Progress...',
+      msg: `Target Path: ${srcSvgDir}`,
+    };
+    log(alert);
+
+    await fs_extra.emptyDir(distSvgDir);
+    const srcSvgData = await getSvgs(srcSvgDir, distSvgDir);
+    await optimizeSvgs(srcSvgData, loadConfigPlugins);
+
+    alert = {
+      type: 'success',
+      name: 'DONE',
+      msg: `Optimized SVGs moved to ${distSvgDir}`,
+    };
+    log(alert);
+  } catch (e) {
+    console.error(e);
+    process.exit(1);
+  }
+}
+async function optimizeSvgs(srcSvgData, loadConfigPlugins) {
+  let pluginOptions = [];
+  loadConfigPlugins.plugins.forEach(function (data) {
+    for (key in data) {
+      if (data[key]) pluginOptions = [...pluginOptions, data];
+    }
+  });
+  const optimizePass = new svgo({
+    plugins: [...pluginOptions],
+  });
+
+  await Promise.all(
+    srcSvgData.map(async (svgData) => {
+      return optimizeSvg(optimizePass, svgData);
+    })
+  );
+}
+async function optimizeSvg(optimizePass, svgData) {
+  const srcSvgContent = await fs_extra.readFile(svgData.srcFilePath, 'utf8');
+  const optimizedSvg = await optimizePass.optimize(srcSvgContent, {
+    path: svgData.srcFilePath,
+  });
+  const optimizedCode = optimizedSvg.data;
+
+  svgData.optimizedSvgContent = optimizedCode;
+
+  await fs_extra.writeFile(svgData.distSvgFilePath, optimizedSvg.data);
+}
+
+async function getSvgs(srcSvgDir, distSvgDir) {
+  const svgFiles = (await fs_extra.readdir(srcSvgDir)).filter((fileName) => {
+    return !fileName.startsWith('.') && fileName.endsWith('.svg');
+  });
+  const svgData = await Promise.all(
+    svgFiles.map(async (fileName) => {
+      if (fileName.toLowerCase() !== fileName) {
+        throw new Error(`svg filename "${fileName}" must be all lowercase`);
+      }
+      const srcFilePath = path.join(srcSvgDir, fileName);
+      const distSvgFilePath = path.join(distSvgDir, fileName);
+      const dotSplit = fileName.split('.');
+      if (dotSplit.length > 2) {
+        throw new Error(
```

Comment:
How is the error handled in this application ?

Reply:
Centralized level in build block, Tested. I cannot add try-catch{} in all functions as it will result in eslint error: 'Unnecessary try-catch wrapper '. So done at the central level.(fn)build.

Reply:
Fixed